study_group/models.py

```diff
@@ -0,0 +1,53 @@
+from django.db import models
+from django.db.models import Q
+
+from users.models import Profile
+
+
+class GroupManager(models.Manager):
+    def get_all_non_full_groups(self):
+        return self.annotate(members_count=models.Count("group_members")) \
+            .filter(members_count__lt=models.F('capacity'))
+
+    def search_group_by_keyword(self, keyword: str):
+        return self.filter(Q(group_description__icontains=keyword) | Q(field__icontains=keyword))
+
+
+class StudyGroup(models.Model):
+    study_group = models.AutoField(primary_key=True)
+    owner_id = models.ForeignKey(Profile, on_delete=models.CASCADE)
+    field = models.CharField(max_length=30)
+    group_description = models.TextField()
+    capacity = models.PositiveIntegerField()
+    objects = GroupManager()
+
+    def join_group(self, profile: Profile):
+        if self.is_group_full():
+            raise ValueError("Group is already full")
+
+        group_member = GroupMember(group_id=self, private_id=profile)
+        group_member.save()
+
+    def is_group_full(self):
+        return self.group_members.count() >= self.capacity
+
+    def get_all_group_members(self):
+        return self.group_members.all()
+
+    # create a study group and add creator to group
+    @classmethod
+    def create(cls, **kwargs):
+        group_owner = kwargs.pop('group_owner', None)
+        if group_owner is None:
```

Comment:
The default is [`blank=False`](https://docs.djangoproject.com/en/4.2/ref/models/fields/#blank), do you still need this check?

Reply:
No need for this check